interpreter/cling/lib/Interpreter/DeclUnloader.cpp

```diff
@@ -345,6 +345,22 @@ namespace {
     }
   };
 
+  template <class EntryType>
+  void removeSpecializationImpl(llvm::FoldingSetVector<EntryType>& Specs,
+                                const EntryType* Entry) {
+    llvm::SmallVector<EntryType*, 4> Filtered;
```

Comment:
"Filtered" doesn't tell me whether it's the gold you want to keep, or the coffee ground, i.e. do you throw away the filtered or are they the ones to survive. IIUC the latter is the case here. So I'd recommend to rename the variable to
```suggestion
    // Remove only Entry from Specs. Keep all others.
    llvm::SmallVector<EntryType*, 4> Keep;
```